diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/chapter3/files/FileStructureRule.kt

```diff
@@ -188,22 +197,72 @@ class FileStructureRule(private val configRules: List<RulesConfig>) : Rule("file
                 }
             }
             .forEach { FILE_WILDCARD_IMPORTS.warn(configRules, emitWarn, isFixMode, it.text, it.startOffset, it) }
-
         val sortedImportsGroups = if (importsGroupingConfig.useRecommendedImportsOrder) {
             regroupImports(imports.map { it.psi as KtImportDirective })
                 .map { group -> group.map { it.node } }
         } else {
             listOf(imports)
         }
             .map { group -> group.sortedBy { it.text } }
-
         if (sortedImportsGroups.flatten() != imports) {
             FILE_UNORDERED_IMPORTS.warnAndFix(configRules, emitWarn, isFixMode, "${sortedImportsGroups.flatten().first().text}...", node.startOffset, node) {
                 rearrangeImports(node, imports, sortedImportsGroups)
             }
         }
     }
 
+    @Suppress("UnsafeCallOnNullableType")
+    private fun checkUnusedImport(
+        node: ASTNode
+    ) {
+        findAllReferences(node)
+        packageName = (node.findChildByType(PACKAGE_DIRECTIVE)?.psi as KtPackageDirective).qualifiedName
+        node.findChildByType(IMPORT_LIST)?.getChildren(TokenSet.create(IMPORT_DIRECTIVE))?.toList()
+            ?.forEach { import ->
+                val ktImportDirective = import.psi as KtImportDirective
+                val importName = ktImportDirective.importPath?.importedName?.asString()
+                val importPath = ktImportDirective.importPath?.pathStr!!  // importPath - ifNOtParsed & Nullable
+                if (ktImportDirective.aliasName == null &&
+                        packageName.isNotEmpty() && importPath.startsWith("$packageName.") &&
+                        importPath.substring(packageName.length + 1).indexOf('.') == -1
+                ) {
+                    // this branch corresponds to imports from the same package
+                    deleteImport(importPath, node, ktImportDirective)
+                } else if (importName != null && !refSet.contains(
+                    importName
+                )
+                ) {
+                    // this import is not used anywhere
+                    deleteImport(importPath, node, ktImportDirective)
+                }
+            }
+    }
+
+    private fun deleteImport(
+        importPath: String,
+        node: ASTNode,
+        ktImportDirective: KtImportDirective
+    ) {
+        UNUSED_IMPORT.warnAndFix(
+            configRules, emitWarn, isFixMode,
+            "$importPath - unused import",
+            node.startOffset, node
+        ) { ktImportDirective.delete() }
+    }
+
+    private fun findAllReferences(node: ASTNode) {
+        node.findAllNodesWithSpecificType(OPERATION_REFERENCE)?.forEach { ref ->
+            if (!ref.isPartOf(IMPORT_DIRECTIVE)) {
+                operatorMap.filterValues { it == ref.text }.keys.forEach { key -> refSet.add(key) }
```

Comment:
```suggestion
                operatorMap.filterValues { it == ref.text }.keys.singleOrNull()?.let { key -> refSet.add(key) }
```
I believe mapping should be one-to-one here?

Reply:
The same operators can have different keys. 
"unaryPlus" to "+"
"plus" to "+"

Reply:
Oh, ok, makes sense.